HEInventions.Logging/RavenLog.cs

```diff
@@ -0,0 +1,219 @@
+    /// to be run at a time.
+    /// </summary>
+    public sealed class RavenLog
+    {
+        #region Constants
+        /// <summary>
+        /// If we cannot retrieve the machine settings, set a default value.
+        /// </summary>
+        public const string UndefinedSetting = "UNDEFINED";
+        #endregion
+
+        #region Properties and Fields
+        /// <summary>
+        /// Raven client for Sentry logging.
+        /// </summary>
+        private RavenClient ravenClient;
+
+        /// <summary>
+        /// Store the host ID from the data in the settings file.
+        /// </summary>
+        public string InstallationHostID = "";
+
+        /// <summary>
+        /// Only send sentry events that equal to, or are above, the defined threshold.
+        /// </summary>
+        public string ErrorThreshold = "";
+
+        /// <summary>
+        /// Triggered when a Sentry event is sent to the server.
+        /// </summary>
+        public event Action<SentryEvent> OnMessage;
+
+        /// <summary>
+        /// Stores the RavenLog instance state.
+        /// </summary>
+        private static RavenLog _Instance = null;
+
+        /// <summary>
+        /// Creates a RavenLog instance.
+        /// </summary>
+        public static RavenLog Instance
+        {
+            get
+            {
+                if (_Instance == null)
+                    throw new InvalidOperationException("RavenLog must be configured before it can be used.");
+                return _Instance;
+            }
+            private set
+            {
+                _Instance = value;
+            }
+        }
+        #endregion
+
+        #region RavenLog Configuration.
+        /// <summary>
+        /// Configures RavenLog and creates a new instance by setting the Sentry DSN, a severity threshold, 
+        /// specifying a settings file e.g. to configure the hostID, and which keys to look for in the settings file.
+        /// </summary>
+        /// <param name="sentryDSN">The Sentry server to post events to.</param>
+        /// <param name="errorThreshold">The severity threshold - all errors equal to or above this will be logged.</param>
+        /// <param name="settingsPath">Location of a settings file (typically JSON) which allows a hostID to be sent with 
+        /// Sentry events.</param>
+        /// <param name="configKeys">The keys to extract values from in the settings file.</param>
+        public static void Configure(string sentryDSN, string errorThreshold, string settingsPath = "", StringCollection configKeys = null)
+        {
+            if (_Instance != null)
+                throw new InvalidOperationException("RavenLog is already configured.");
+
+            Instance = new RavenLog()
+            {
+                ravenClient = new RavenClient(sentryDSN),
+                ErrorThreshold = errorThreshold,
+                InstallationHostID = (settingsPath != "" && configKeys != null) ? 
```

Comment:
`settingsPath != ""`  -->  `String.IsNullOrWhitespace`

Reply:
Why validate just this one?  Defaults can be set in the parameter definitions if you want.

Reply:
Yeah good point, done